extensions/shared/external-media/sources/with-media.js

```diff
@@ -53,6 +54,45 @@ export default function withMedia() {
 				};
 			}
 
+			componentDidMount() {
+				this.removeArrowKeysPropagationHandler = this.attachArrowKeysPropagationHandler();
+			}
+
+			componentWillUnmount() {
+				if ( this.removeArrowKeysPropagationHandler ) {
+					this.removeArrowKeysPropagationHandler();
+				}
+			}
+
+			attachArrowKeysPropagationHandler = () => {
+				const eventListenerEl = document.querySelector( '.components-modal__content' );
+				if ( ! eventListenerEl ) {
+					return null;
+				}
+
+				eventListenerEl.addEventListener( 'keydown', this.stopArrowKeysPropagation );
+				return () => {
+					eventListenerEl.removeEventListener( 'keydown', this.stopArrowKeysPropagation );
+				};
+			};
```

Comment:
It took me awhile to understand what was happening here. I think it would be easier to understand with:
- more comments (`attachArryKeysPropagationHandler` returning the `removeEventListener` for itself threw me for a minute)
- Restructuring the namings. Same as above, `this.removeArrowKeysPropagationHandler = this.attachArrowKeysPropagationHandler();` is a strange line. 

If this is an accepted way of going about this pattern, then we can leave it. It's my first time seeing this structure. It does work quite well though!

Reply:
That's a convention that's used i.e. [in the `useEffect` hook](https://reactjs.org/docs/hooks-effect.html#effects-without-cleanup), where you can return a `cleanup` function that will be executed when the component unmounts. I don't think there's a name for it, though, but basically, it's a subscriber function that returns its own unsubscriber. This is also a good example:

```js
const observable = new Observable(function subscribe(subscriber) {
  // Keep track of the interval resource
  const intervalId = setInterval(() => {
    subscriber.next('hi');
  }, 1000);

  // Provide a way of canceling and disposing the interval resource
  return function unsubscribe() {
    clearInterval(intervalId);
  };
});
```

Source: https://rxjs.dev/guide/observable#disposing-observable-executions

Reply:
Could we do something similar to how they have it set-up in the Observable example?

`attachArrowKeysPropagationHandler` -> `arrowKeysPropagationHandler`

`return () => ` -> `return remove() => `

So, something like:
```
componentDidMount() {
  this.removeArrowKeysPropagationHandler = this.arrowKeysPropagationHandler.remove();
}

arrowKeysPropagationHandler = () => {
  // ... handle adding the listener

  // give the anonymous function a name
  return remove() => {
    ...
  }
}
```

Reply:
No need to use lifecycle calls actually :D Attached it directly here: https://github.com/Automattic/jetpack/pull/16055/commits/64f43198c7cca9fdcd4f726596da9bc35526ce04